util/election/runner.go

```diff
@@ -0,0 +1,187 @@
+// Copyright 2017 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package election
+
+import (
+	"context"
+	"math/rand"
+	"time"
+
+	"github.com/golang/glog"
+	"github.com/google/trillian/util"
+)
+
+const (
+	// Minimum values for configuration intervals.
+	MinPreElectionPause    = 10 * time.Millisecond
+	MinMasterCheckInterval = 50 * time.Millisecond
+	MinMasterHoldInterval  = 10 * time.Second
+)
+
+// RunnerConfig describes the parameters for an election Runner.
+type RunnerConfig struct {
+	// PreElectionPause is the maximum interval to wait before starting a
+	// mastership election for a particular log.
+	PreElectionPause time.Duration
+	// MasterCheckInterval is the interval between checks that we still
+	// hold mastership for a log.
+	MasterCheckInterval time.Duration
+	// MasterHoldInterval is the minimum interval to hold mastership for.
+	MasterHoldInterval time.Duration
+	// ResignOdds gives the chance of resigning mastership after each
+	// check interval, as the N for 1-in-N.
+	ResignOdds int
+
+	TimeSource util.TimeSource
+}
+
+// fixupRunnerConfig ensures operation parameters have required minimum values.
+func fixupRunnerConfig(cfg *RunnerConfig) {
```

Comment:
I think that these values should be application- and/or (MasterElection)implementation- specific.
It's fine having them here though, because of this PR's purpose.